osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableHitStrong.cs

```diff
@@ -46,18 +44,26 @@ protected override void CheckJudgement(bool userTriggered)
                 Judgement.SecondHit = true;
         }
 
-        protected override bool HandleKeyPress(Key key)
+        public override bool OnReleased(TaikoAction action)
+        {
+            if (action == firstHitAction)
+                firstKeyHeld = false;

```

Comment:
That's the problem, imagine two strong hits occurring one-after-another. The first hit, then released, but then both of them should get OnReleased, no?